src/python-common/ceph/tests/test_service_spec.py

```diff
@@ -233,14 +235,105 @@ def test_yaml():
   objectstore: bluestore
   wal_devices:
     model: NVME-QQQQ-987
-"""
+---
+service_type: alertmanager
+service_name: alertmanager
+spec:
+  port: 1234
+  user_data:
+    default_webhook_urls:
+    - foo
+---
+service_type: grafana
+service_name: grafana
+spec:
+  port: 1234
+---
+service_type: ingress
+service_id: rgw.foo
+service_name: ingress.rgw.foo
+placement:
+  hosts:
+  - host1
+  - host2
+  - host3
+spec:
+  backend_service: rgw.foo
+  frontend_port: 8080
+  monitor_port: 8081
+  virtual_ip: 192.168.20.1/24
+---
+service_type: nfs
+service_id: mynfs
+service_name: nfs.mynfs
+spec:
+  port: 1234
+---
+service_type: iscsi
+service_id: iscsi
+service_name: iscsi.iscsi
+networks:
```

Comment:
```suggestion
service_name: iscsi.iscsi
placement: {}
networks:
```
test wants an empty placement dir to compare with the yaml.dump() output, but it's valid to omit the placement spec when generating a service spec....

maybe yaml.dump() should omit the placement key when empty?

Reply:
```
>       assert yaml.dump(object) == y
E       AssertionError: assert 'service_type...  foo: /foo\n' == 'service_type...  foo: /foo\n'
E         Skipping 74 identical leading characters in diff, use -v to show
E           llo-world
E         + placement: {}
E           spec:
E             args:
E             - --foo
E             bind_mounts:...
E         
E         ...Full output truncated (26 lines hidden), use '-vv' to show

ceph/tests/test_service_spec.py:321: AssertionError
```

@sebastian-philipp could you please consider Michael's suggestion?

Reply:
I did in #42905 but I need to rethink this more it seems